packages/core/src/modules/proofs/protocol/v2/V2ProofService.ts

```diff
@@ -643,6 +612,53 @@ export class V2ProofService extends ProofService {
     return proofRecord
   }
 
+  public async createProofRequestFromProposal(options: ProofRequestFromProposalOptions): Promise<ProofRequestFormats> {
+    const proofRecordId = options.proofRecord.id
+    const proposalMessage = await this.didCommMessageRepository.findAgentMessage({
+      associatedRecordId: proofRecordId,
+      messageClass: V2ProposalPresentationMessage,
+    })
+
+    if (!proposalMessage) {
+      throw new AriesFrameworkError(`Proof record with id ${proofRecordId} is missing required presentation proposal`)
+    }
+
+    const proposalAttachments = proposalMessage.getAttachmentFormats()
+
+    let result = {}
+
+    for (const attachmentFormat of proposalAttachments) {
+      const service = this.getFormatServiceForFormat(attachmentFormat.format)
+
+      if (!service) {
+        throw new AriesFrameworkError('No format service found for getting requested.')
+      }
+
+      result = {
+        ...result,
+        ...(await service.createProofRequestFromProposal({
+          presentationAttachment: attachmentFormat.attachment,
+        })),
+      }
+    }
+
+    return result
+  }
+
+  public async shouldAutoRespondToProposal(proofRecord: ProofRecord): Promise<boolean> {
+    const proposal = await this.didCommMessageRepository.findAgentMessage({
+      associatedRecordId: proofRecord.id,
+      messageClass: V2ProposalPresentationMessage,
+    })
+
+    if (!proposal) {
+      return false
+    }
+
+    await MessageValidator.validate(proposal)
+    return true
```

Comment:
We should check here whether the proposal matches the request

Reply:
DONE